systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/controllers/StrimziPodSetUtils.java

```diff
@@ -52,6 +53,82 @@ public static void waitForStrimziPodSetLabelsDeletion(String namespaceName, Stri
         }
     }
 
+    /**
+     * Waits until the single specified annotation of a Pod changes its value.
+     *
+     * @param namespaceName    the namespace in which the Pod is located.
+     * @param podName          the name of the Pod whose annotation value needs to be checked.
+     * @param strimziPodSetName the name of the SPS whose Pod's annotation is being observed.
+     * @param annotationKey    the key of the annotation whose value needs to be checked.
+     * @param annotationValue  the expected value of the annotation to change from.
+     */
+    public static void waitUntilPodAnnotationChange(String namespaceName, String podName, String strimziPodSetName, String annotationKey, String annotationValue) {
+
+        LOGGER.info("Waiting for Pod: {}/{} to change annotation value: {}-{}", namespaceName, podName, annotationKey, annotationValue);
+        final ObjectMapper objectMapper = new ObjectMapper();
+
+
+        TestUtils.waitFor("Pod to change value of annotation" + annotationKey + " with key " + annotationKey, Constants.POLL_INTERVAL_FOR_RESOURCE_READINESS,
+            DELETION_TIMEOUT, () -> {
+                // parse pod from SPS
+                StrimziPodSet sps = StrimziPodSetResource.strimziPodSetClient().inNamespace(namespaceName).withName(strimziPodSetName).get();
+                var podMap = sps.getSpec().getPods();
```

Comment:
I really don't like using the `var`, when you can determine the type.
I guess that the `sps.getSpec().getPods()` will return `List<Pod>` right? That's better for filtering and you don't need to do all the conversion.